readthedocs/analytics/models.py

```diff
@@ -33,60 +58,84 @@ class PageView(models.Model):
         verbose_name=_('Version'),
         related_name='page_views',
         on_delete=models.CASCADE,
+        null=True,
     )
     path = models.CharField(max_length=4096)
     view_count = models.PositiveIntegerField(default=0)
     date = models.DateField(default=datetime.date.today, db_index=True)
+    status = models.PositiveIntegerField(
+        default=200,
+        help_text=_("HTTP status code"),
+    )
+
+    objects = PageViewManager()
 
     class Meta:
-        unique_together = ("project", "version", "path", "date")
+        unique_together = ("project", "version", "path", "date", "status")
```

Comment:
It was decided to track them separately to avoid that problem, once the URL has been fixed the number of views shouldn't increase https://github.com/readthedocs/readthedocs.org/issues/8849#issuecomment-1069636664.

Reply:
I don't follow how that comment answers my question 🤔 

I'm talking more about the UX than a technical aspect. As a user, I would see some URLs listed under "Broken links" that are not broken and that doesn't look good to me. So, I'm saying that we should do something on the backend to avoid that scenario.

Reply:
Gotcha! It makes sense and it's better UX 💯